modules/core/arrow-test/src/main/kotlin/arrow/test/laws/ContravariantLaws.kt

```diff
@@ -0,0 +1,34 @@
+package arrow.test.laws
+
+import arrow.Kind
+import arrow.core.compose
+import arrow.test.generators.genConstructor
+import arrow.test.generators.genFunctionAToB
+import arrow.typeclasses.Contravariant
+import arrow.typeclasses.Eq
+import io.kotlintest.properties.Gen
+import io.kotlintest.properties.forAll
+
+object ContravariantLaws {
+
+    inline fun <F> laws(CF: Contravariant<F>, noinline cf: (Int) -> Kind<F, Int>, EQ: Eq<Kind<F, Int>>): List<Law> =
+        listOf(
```

Comment:
In here ContravariantLaws imply InvariantLaws also pass, so replace this line with:

InvariantLaws.laws(...) + listOf(